lib/run-contentful-export.js

```diff
@@ -22,6 +23,45 @@ export default function runContentfulExport (usageParams) {
     spaceId: clients.source.spaceId
   })
     .then((response) => {
+      if (opts.downloadAssets) {
+        var successCount = 0
+        var warningCount = 0
+        var errorCount = 0
+
+        log.info('Downloading ' + response.assets.length + ' assets')
+
+        var arrayOfDownloadPromises = []
+        response.assets.forEach((asset) => {
+          if (asset.fields.file) {
+            Object.keys(asset.fields.file).forEach((key) => {
+              arrayOfDownloadPromises.push(
+                downloadAsset(asset.fields.file[key].url, opts.exportDir).then(
+                  (downLoadedFile) => {
+                    log.info('-> ' + downLoadedFile)
+                    successCount++
+                  },
+                  (error) => {
+                    log.error('-> error downloading ' + asset.fields.file[key].url + ' => ' + error)
+                    errorCount++
+                  }
+                )
+              )
+            })
+          } else {
+            log.warn('-> asset has no file(s)', jsonStringifySafe(asset))
+            warningCount++
+          }
+        })
+
+        Promise.all(arrayOfDownloadPromises).then(
```

Comment:
a neat trick is when creating the promise you do `.catch` and inside the catch `return Promise.resolve({downloaded: false, url:"assetsURL"})` for example this way the Promise.all will always work, you can filter the results further more in `Promise.all().then()` to return the correct amount of assets downloaded